src/view_element.cpp

```diff
@@ -420,11 +420,23 @@ void View::DrawBarLine(DeviceContext *dc, LayerElement *element, Layer *layer, S
         barLine->SetEmptyBB();
         return;
     }
+    StaffDef *drawingStaffDef = staff->m_drawingStaffDef;
+    // Determine the method
+    assert(drawingStaffDef);
+    auto [hasMethod, method] = barLine->GetMethod(drawingStaffDef);
+    if (barLine->HasMethod()) {
```

Comment:
It is not very clear why you need to reassign `method`. Could you add a comment explaining why?

Reply:
The `GetMethod(staffDef)` is overruling libmei's `GetMethod()`, so here it's first looking for the bar method on staffDef and then for the local method.